### PR TITLE
Login register

### PR DESCRIPTION
**Pull request by CShum28**

### CODE REVIEW

File: src/app/components/SignUp/SignUp.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import { toast, ToastContainer } from 'react-toastify';
+import { z } from 'zod';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Button } from '@lib/Button/Button';
+import { Checkbox } from '@lib/Checkbox/Checkbox';
+import { Form, FormControl, FormField, FormItem, FormMessage } from '@lib/Form/Form';
+import { Input } from '@lib/Input/Input';
+
+import 'react-toastify/dist/ReactToastify.css';
```
Comment (Yzma) [inline src/app/components/SignUp/SignUp.tsx]: put this in app.tsx

File: src/app/components/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      // check passwords match
+      toast.error('Passwords do not match');
+    } else {
+      // console.log(values);
+    }
+  }
+
+  return (
+    <div className="mt-[60px] flex max-w-[400px] flex-col items-center rounded-md bg-[#fff] p-10">
+      <ToastContainer />
```
Comment (Yzma) [inline src/app/components/SignUp/SignUp.tsx]: this as well

File: src/app/components/SignUp/SignUp.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+          {/* {modal && <p className="text-[14px] text-[rgb(237,62,62)]">Please agree to our terms of service</p>} */}
+
+          <Button size="loginSignUp" type="submit" variant="loginSignUp">
+            Sign up
+          </Button>
+        </form>
+      </Form>
+
+      <div className="mt-[80px] flex flex-row text-[14px]">
+        <a className="flex flex-row text-[#8f99a1] hover:text-[#3db4f2]" href="/login">
```
Comment (Yzma) [inline src/app/components/SignUp/SignUp.tsx]: Use the Link component from react-router

File: src/app/components/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+import { Form, FormControl, FormField, FormItem, FormMessage } from '@lib/Form/Form';
+import { Input } from '@lib/Input/Input';
+
+import 'react-toastify/dist/ReactToastify.css';
+
+const formSchema = z.object({
+  email: z.string().min(2, { message: 'Please enter your email address' }).max(50),
+  username: z.string().min(2, { message: 'Name must be atleast 2 characters' }).max(50),
+  password: z.string().min(2, { message: 'Password be atleast 2 characters' }).max(50),
+  confirmPassword: z.string().min(2, { message: 'Confirm password must be atleast 2 characters' }).max(50),
```
Comment (Yzma) [inline src/app/components/SignUp/SignUp.tsx]: - Add 'tos' into schema
- Add logic to say when it fails show the toast
- Remove the checkbox useState

File: modules/gamelist-ui/src/lib/Button/Button.tsx [hunk 5]
```diff
@@ -14,12 +14,14 @@ const buttonVariants = cva(
         secondary: 'bg-secondary text-secondary-foreground hover:bg-secondary/80',
         ghost: 'hover:bg-accent hover:text-accent-foreground',
         link: 'text-primary underline-offset-4 hover:underline',
+        loginSignUp: 'text-white bg-[#3db4f2] rounded-[3px] mt-[10px] cursor-pointer font-semibold',
```
Comment (Yzma) [inline modules/gamelist-ui/src/lib/Button/Button.tsx]: remove arbitrary values

File: src/app/components/SignUp/SignUp.tsx (hunk 1 above)
Comment (CShum28) [reply src/app/components/SignUp/SignUp.tsx]: Updated this on main.tsx instead

File: src/app/components/SignUp/SignUp.tsx (hunk 2 above)
Comment (CShum28) [reply src/app/components/SignUp/SignUp.tsx]: updated this on main.tsx instead

File: src/app/components/SignUp/SignUp.tsx (hunk 3 above)
Comment (CShum28) [reply src/app/components/SignUp/SignUp.tsx]: updated to link

File: src/app/components/SignUp/SignUp.tsx (hunk 4 above)
Comment (CShum28) [reply src/app/components/SignUp/SignUp.tsx]: Updated

File: modules/gamelist-ui/src/lib/Button/Button.tsx (hunk 5 above)
Comment (CShum28) [reply modules/gamelist-ui/src/lib/Button/Button.tsx]: done

**Pull request closed without merging**